cardano-cli/src/Cardano/CLI/EraBased/Options/Address.hs

```diff
@@ -13,23 +13,34 @@ import           Cardano.CLI.Environment (EnvCli (..))
 import           Cardano.CLI.EraBased.Commands.Address
 import           Cardano.CLI.EraBased.Options.Common
 
-import           Data.Foldable
 import           Options.Applicative hiding (help, str)
 import qualified Options.Applicative as Opt
 
-pAddressCmds :: CardanoEra era -> EnvCli -> Parser (AddressCmds era)
+pAddressCmds :: ()
+  => CardanoEra era
+  -> EnvCli
+  -> Maybe (Parser (AddressCmds era))
 pAddressCmds _ envCli =
```

Comment:
So we're not using the `CardanoEra` value here because these commands are constant for all the eras. I'm assuming you're doing this in the event that e.g the cryptography changes in a future era.

Reply:
Or if we add new commands that are only apply to a future era.

Reply:
Some scenarios that might cause the era to be need:

* A new command or flag added that only apply to newer eras
* An existing command or flag needs to be deleted for newer eras